src/writer/async_tokio.rs

```diff
@@ -82,6 +84,178 @@ impl<W: AsyncWrite + Unpin> Writer<W> {
+
+    /// writer
+    ///     .create_element("paired")
+    ///     .with_attribute(("attr1", "value1"))
+    ///     .with_attribute(("attr2", "value2"))
+    ///     .write_text_content_async(BytesText::new("text"))
+    ///     .await
+    ///     .expect("failure");
+
+    /// tokio_buffer.flush().await.expect("flush failed");
+
+    /// assert_eq!(
+    ///     std::str::from_utf8(&buffer).unwrap(),
+    ///     r#"<paired attr1="value1" attr2="value2">text</paired>"#
+    /// );
+    /// }
+    pub async fn write_text_content_async(self, text: BytesText<'_>) -> Result<&'a mut Writer<W>> {
+        self.writer
+            .write_event_async(Event::Start(self.start_tag.borrow()))
+            .await?;
+        self.writer.write_event_async(Event::Text(text)).await?;
+        self.writer
+            .write_event_async(Event::End(self.start_tag.to_end()))
+            .await?;
+        Ok(self.writer)
+    }
+
+    /// Write a CData event `<![CDATA[...]]>` inside the current element.
+    pub async fn write_cdata_content_async(
+        self,
+        text: BytesCData<'_>,
+    ) -> Result<&'a mut Writer<W>> {
+        self.writer
+            .write_event_async(Event::Start(self.start_tag.borrow()))
+            .await?;
+        self.writer.write_event_async(Event::CData(text)).await?;
+        self.writer
+            .write_event_async(Event::End(self.start_tag.to_end()))
+            .await?;
+        Ok(self.writer)
+    }
+
+    /// Write a processing instruction `<?...?>` inside the current element.
+    pub async fn write_pi_content_async(self, text: BytesText<'_>) -> Result<&'a mut Writer<W>> {
+        self.writer
+            .write_event_async(Event::Start(self.start_tag.borrow()))
+            .await?;
+        self.writer.write_event_async(Event::PI(text)).await?;
+        self.writer
+            .write_event_async(Event::End(self.start_tag.to_end()))
+            .await?;
+        Ok(self.writer)
+    }
+
+    /// Write an empty (self-closing) tag.
+    /// # Example
+    ///
+    /// ```rust
+    /// # use quick_xml::Result;
+    /// # use quick_xml::Writer;
+    /// # use quick_xml::events::BytesText;
+    /// # use tokio::io::AsyncWriteExt;
+    /// # #[tokio::main(flavor = "current_thread")] async fn main() {
+    ///    let mut buffer = Vec::new();
+    ///    let mut tokio_buffer = tokio::io::BufWriter::new(&mut buffer);
+    ///    let mut writer = Writer::new_with_indent(&mut tokio_buffer, b' ', 4);
+    ///
+    ///    writer
+    ///        .create_element("empty")
+    ///        .with_attribute(("attr1", "value1"))
+    ///        .with_attribute(("attr2", "value2"))
+    ///        .write_empty_async()
+    ///        .await
+    ///        .expect("failure");
+    ///
+    ///     tokio_buffer.flush().await.expect("flush failed");
+    ///
+    ///     assert_eq!(
+    ///      std::str::from_utf8(&buffer).unwrap(),
+    ///      r#"<empty attr1="value1" attr2="value2"/>"#
+    ///     );
+    /// }
```

Comment:
```suggestion
    /// # }
```
And also please dedent the visible code in example